apps/agent/scripts/buidler-hooks.js

```diff
@@ -0,0 +1,68 @@
+let voting, accounts
+
+const ZERO_ADDRESS = '0x0000000000000000000000000000000000000000'
+
+module.exports = {
+  postDao: async function ({ _experimentalAppInstaller, log }, bre) {
+    const bigExp = (x, y) =>
+      bre.web3.utils
+        .toBN(x)
+        .mul(bre.web3.utils.toBN(10).pow(web3.utils.toBN(y)))
+    const pct16 = (x) => bigExp(x, 16)
+
+    // Retrieve accounts.
+    accounts = await bre.web3.eth.getAccounts()
+
+    // Deploy a minime token an generate tokens to root account
+    const minime = await _deployMinimeToken(bre)
+    await minime.generateTokens(accounts[1], pct16(10))
+    log(`> Minime token deployed: ${minime.address}`)
+
+    const tokens = await _experimentalAppInstaller('token-manager', {
+      skipInitialize: true,
+    })
+
+    await minime.changeController(tokens.address)
+    log(`> Change minime controller to tokens app`)
+    await tokens.initialize([minime.address, true, 0])
+    log(`> Tokens app installed: ${tokens.address}`)
+
+    voting = await _experimentalAppInstaller('voting', {
+      initializeArgs: [
+        tokens.address,
+        pct16(50), // support 50%
+        pct16(25), // quorum 15%
+        604800, // 7 days
+      ],
+    })
+    log(`> Voting app installed: ${voting.address}`)
+
+    await tokens.createPermission('MINT_ROLE', voting.address)
+    await voting.createPermission('CREATE_VOTES_ROLE', tokens.address)
```

Comment:
I don't think this quite works the way we think it does—these role strings should be hashed.

Reply:
We are handling that logic inside the plugin to make it more friendly to the user.